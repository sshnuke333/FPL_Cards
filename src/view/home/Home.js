import React from 'react';
import { useSelector } from 'react-redux';
import { headerSelector } from '../../store/Header.slice';
import { Div, ShowCase, StyledP } from './Home.styles';
import { Rules } from './Rules';
import { Card } from '../../components/Card';

export const Home = () => {
    const { fplPlayers } = useSelector(headerSelector);
    return (
        <>
            <Div>
                <ShowCase>
                    {Object.keys(fplPlayers).length !== 0 ? (
                        <>
                            <Card display="none" player={fplPlayers[32]} />
                            <Card
                                margin="-5rem"
                                zIndex="1000"
                                player={fplPlayers[371]}
                            />
                            <Card
                                margin="-5rem"
                                display="none"
                                player={fplPlayers[404]}
                            />
                        </>
                    ) : (
                        <div></div>
                    )}
                </ShowCase>
                <StyledP>A fun fpl based card game</StyledP>
            </Div>
            <Rules player={fplPlayers[18]} />
        </>
    );
};
