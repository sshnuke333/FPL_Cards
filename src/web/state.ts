import { configureStore } from '@reduxjs/toolkit';

import HeaderReducer from './store/Header.slice';
import cardReducer from './store/Cards.slice';

export const store = configureStore({
    reducer: {
        Header: HeaderReducer,
        cards: cardReducer,
    },
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;
