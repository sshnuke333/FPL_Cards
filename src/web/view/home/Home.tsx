import React from 'react';
import { useSelector } from 'react-redux';
import { headerSelector } from '../../store/Header.slice';
import { Div, ShowCase, StyledP } from './Home.styles';
import Rules from './Rules/Rules';
import { Card } from '../../components';

const Home = () => {
    const { fplPlayers } = useSelector(headerSelector);
    return (
        <>
            <Div>
                <ShowCase>
                    {Object.keys(fplPlayers).length !== 0 ? (
                        <>
                            <Card display="none" player={fplPlayers[503]} />
                            <Card
                                margin="-5rem"
                                zIndex="1000"
                                player={fplPlayers[446]}
                            />
                            <Card
                                margin="-5rem"
                                display="none"
                                player={fplPlayers[5]}
                            />
                        </>
                    ) : (
                        <div></div>
                    )}
                </ShowCase>
                <StyledP>A fun fpl based card game</StyledP>
            </Div>
            <Rules player={fplPlayers[11]} />
        </>
    );
};

export default Home;
