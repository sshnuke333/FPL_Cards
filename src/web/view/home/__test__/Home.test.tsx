import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../state';

import Home from '../Home';

describe('Home', () => {
    it('renders successfully', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <Home />
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
