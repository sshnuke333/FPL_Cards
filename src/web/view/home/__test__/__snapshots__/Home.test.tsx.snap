// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders successfully 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe gQyfUl"
  >
    <div
      class="sc-gswNZR fXKJtf"
    >
      <div />
    </div>
    <p
      class="sc-dkrFOg kalUTa"
    >
      A fun fpl based card game
    </p>
  </div>
  <div
    class="sc-hhOBVt cjRwpi"
  >
    <h3
      style="font-size: 5rem;"
    >
      Rules
    </h3>
    <ul
      class="sc-gYbzsP beKSyl"
    >
      <li
        class="sc-cCjUiG NgKqX"
      >
        You will assigned 11 cards by default
      </li>
      <li
        class="sc-cCjUiG NgKqX"
      >
        You can change this using settings in the play area in the next page
      </li>
      <li
        class="sc-cCjUiG NgKqX"
      >
        Only your current card will visible in play area
      </li>
      <li
        class="sc-cCjUiG NgKqX"
      >
        You can peek at your cards
      </li>
      <li
        class="sc-cCjUiG NgKqX"
      >
        Opponent's card will be hidden until you make a choice
      </li>
      <li
        class="sc-cCjUiG NgKqX"
      >
        rest of the opponent cards will be hidden
      </li>
      <li
        class="sc-cCjUiG NgKqX"
      >
        you win the game if you get all opponent's cards
      </li>
      <li
        class="sc-cCjUiG NgKqX"
      >
        your win / lose based on your card stats, check example below
      </li>
    </ul>
    <div
      class="sc-cjibBx bBvVpR"
    >
      <div
        class="sc-cjibBx sc-jcMfQk bBvVpR bsSQtr"
      >
        <p
          class="sc-cabOPr bcscRp"
          color="#dad400"
        >
          Click on the stat button for info
        </p>
        <p
          class="sc-cabOPr cBUyqW"
        />
      </div>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
